src/main/java/eu/ai4work/sws/service/RuleEngineService.java

```diff
@@ -49,54 +43,76 @@ public SlidingDecisionResult applySlidingDecisionRules(Map<String, Object> slidi
+        return fuzzyInferenceSystem;
     }
 
     /**
-     * Determines the linguistic term with the highest membership degree for the given variable.
+     * Maps a fuzzy inference result to its corresponding linguistic term based on the highest membership degree.
      *
      * @param suggestedWorkSharingApproachAsFuzzyVariable The fuzzy output variable to evaluate.
-     * @return The output as linguistic term, i.e. the name of the membership function with the highest membership degree.
+     * @return The output as a linguistic term, i.e., the name of the membership function with the highest membership degree.
+     * @throws UnIdentifiedTermException if no linguistic term can be identified.
      */
-    private String mapFuzzyInferenceResultToLinguisticTerm(Variable suggestedWorkSharingApproachAsFuzzyVariable) {
-        String linguisticTerm = null;
-        // The variable is initialized to -1.0. This value is chosen because membership degrees in fuzzy logic
-        // are typically between 0 and 1. Initializing to -1.0 ensures that any valid membership degree
-        // (which will be greater than -1.0) will replace this initial value.
-        double highestMembershipDegree = -1.0;
-
-        for (Map.Entry<String, LinguisticTerm> termEntry : suggestedWorkSharingApproachAsFuzzyVariable.getLinguisticTerms().entrySet()) {
-            double membershipDegree = termEntry.getValue().getMembershipFunction().membership(suggestedWorkSharingApproachAsFuzzyVariable.getLatestDefuzzifiedValue());
-            if (membershipDegree > highestMembershipDegree) {
-                highestMembershipDegree = membershipDegree;
-                linguisticTerm = termEntry.getKey();
-            }
-        }
+    private String mapFuzzyInferenceResultToLinguisticTerm(Variable suggestedWorkSharingApproachAsFuzzyVariable) throws UnIdentifiedTermException {
+        return suggestedWorkSharingApproachAsFuzzyVariable.getLinguisticTerms().entrySet().stream()
+                // Map each linguistic term to its corresponding membership degree
+                // The key is the linguistic term name
+                // The value is membership degree for the latest defuzzified value
+                .map(linguisticTermWithMembershipDegree -> Map.entry(
+                        linguisticTermWithMembershipDegree.getKey(),
+                        linguisticTermWithMembershipDegree.getValue().getMembershipFunction()
+                                .membership(suggestedWorkSharingApproachAsFuzzyVariable.getLatestDefuzzifiedValue())
+                ))
+                // Identify the linguistic term with the highest membership degree
+                .max(Map.Entry.comparingByValue())
+                // Retrieve the linguistic term with the highest membership degree, if available
+                .map(Map.Entry::getKey)
+                // If no valid term is found, throw an exception
+                .orElseThrow(() -> new UnIdentifiedTermException(
```

Comment:
This exception name is very "generic" and will make it hard for readers to understand the issue at hand, similar for the message below. Could we find a more specific name or at least a more specific message to make readers understand what the issue might be and how to fix it?